src/FactorGraphTypes.jl

```diff
@@ -132,18 +132,24 @@ $(TYPEDEF)
 
 TODO remove Union types -- issue #383
 """
-mutable struct FactorMetadata
-  factoruserdata
-  variableuserdata::Union{Vector, Tuple}
-  variablesmalldata::Union{Vector, Tuple}
-  solvefor::Union{Symbol, Nothing}
-  variablelist::Union{Nothing, Vector{Symbol}}
-  dbg::Bool
-  FactorMetadata() = new() # [], []
-  FactorMetadata(x1, x2::Union{Vector,Tuple},x3) = new(x1, x2, x3, nothing, nothing, false)
-  FactorMetadata(x1, x2::Union{Vector,Tuple},x3,x4::Symbol) = new(x1, x2, x3, x4, nothing, false)
-  FactorMetadata(x1, x2::Union{Vector,Tuple},x3,x4::Symbol,x5::Vector{Symbol};dbg::Bool=false) = new(x1, x2, x3, x4, x5, dbg)
+mutable struct FactorMetadata{T}
+  factoruserdata # deprecate, not in use
```

Comment:
Suggest not deprecating this field.  It is likely used in some of the fancier cases like SAS.  See examples in Caesar.

Reply:
I changed comment accordingly, I don't see its usefulness and can't find it populated anywhere. So its worth a look to remove.